### DISCUSSION TITLE
Google Consent Mode v2?

### DISCUSSION THREAD

**Discussion by younex1** [johannschopplich/nuxt-gtag#48] [Q&A] [answered]
Is it supporting Google Consent Mode v2?

**Comment by johannschopplich** [comment]
Would you mind elaborating, what that is? 🙌

**Comment by younex1** [comment]
Hi Johann,

sure, thank you for your reply.

Google is introducing consent mode V2 which means to sent 2 more parameters beginning from 06.03.2024:
-ad_user_data 
-ad_personalization 

If not it will decrease tracking capabilities.

Ressources:
https://developers.google.com/tag-platform/security/guides/consent?sjid=3562207944374663292-EU&consentmode=advanced#upgrade-consent-v2
https://developers.google.com/tag-platform/security/guides/consent?sjid=3562207944374663292-EU&consentmode=advanced#upgrade-consent-v2

**Reply by johannschopplich** [reply, marked as answer]
Sure thing, you can send any parameters to Google via the `gtag` method:

```ts
const { gtag } = useGtag()

gtag('consent', 'default', {
  'ad_storage': 'denied',
  'ad_user_data': 'denied',
  'ad_personalization': 'denied',
  'analytics_storage': 'denied'
})
```

> [!TIP]
> Head over to the docs for the whole [`useGtag`](https://github.com/johannschopplich/nuxt-gtag?tab=readme-ov-file#usegtag) 
usage.

**Comment by pperzyna** [comment]
the solution gives the error  ` A tag read consent state before a default was set `

See solution: https://github.com/johannschopplich/nuxt-gtag/pull/49
